content/post/golang-api.md

```diff
@@ -0,0 +1,317 @@
+
+func (cw *CountWord) Name() string {
+	return "count word"
+}
+
+func (cw *CountWord) Compute(data interface{}, state interface{}) (interface{}, bool) {
+	word := data.(*string)
+	wordTotals := state.(*WordTotals)
+	wordTotals.Update(*word)
+	return wordTotals.GetCount(*word), true
+}
+```
+
+#### Outgoing Messages and the Encoder
+
+In our example, the outgoing message is represented within the application as an object that stores the word and the count of the number of times that word has been seen in the event stream.
+
+```go
+type WordCount struct {
+	Word string
+	Count uint64
+}
+```
+
+The `Encoder` contains the logic for transforming this object into a list of bytes that will then be sent on the outgoing TCP connection. In the example, outgoing messages are strings of `WORD => COUNT\n` where `WORD` is the word being counted and `COUNT` is the count.
+
+```go
+type Encoder struct {}
+
+func (encoder *Encoder) Encode(data interface{}) []byte {
+	word_count := data.(*WordCount)
+	msg := fmt.Sprintf("%s => %d\n", word_count.Word, word_count.Count)
+	fmt.Println(msg)
+	return []byte(msg)
+}
+```
+
+This example uses a TCP sink, but Wallaroo also supports Kafka sinks. Other types of sinks will be added in the future.
+
+#### Serialization and Deserialization
+
+Wallaroo needs to be able to serialize and deserialize objects in order to store them to disk for resiliency and recovery, and also to send them to other worker nodes when the application is being used in a multi-worker cluster. The developer must create code to do this. We've omitted that code in this blog post, but you can learn more about it in the ["Interworker Serialization and Resilience"](https://docs.wallaroolabs.com/book/go/api/interworker-serialization-and-resilience.html) section of our documentation.
+
+## A Scalable Event Processing Application
+
+This application can run on one worker and can scale horizontally by adding more and more workers. Wallaroo's flexibility makes it easy to adapt to whatever partitioning strategy your application requires. Take a look at our documentation for [information about how to run a Wallaroo cluster](https://docs.wallaroolabs.com/book/appendix/clustering.html).
+
+## Check It Out
+
+If you're interested in running this application yourself, take a look at the [Wallaroo documentation](https://docs.wallaroolabs.com) and the [word count example application](https://github.com/WallarooLabs/wallaroo/tree/release/examples/go/word_count) that we've built. You'll find instructions on setting up Wallaroo and running applications. And take a look at our [community page](https://www.wallaroolabs.com/community) to sign up for our mailing list or join our IRC channel to ask any question you may have.
+
+You can also watch this video to see Wallaroo in action. Our VP of Engineering walks you through the concepts that were covered in this blog post and then shows the word count application scaling by adding new workers to the cluster.
```

Comment:
Should the video note that its using the old Python API but the core concepts still hold?

Reply:
I'll change the wording to make that clear.